frontend/cypress/unit/FakeDataSchemaService.test.js

```diff
@@ -0,0 +1,241 @@
+import FakeDataSchemaService from '../../src/services/FakeDataSchemaService';
+import CancellationError from '../../src/utils/CancellationError';
+import { CancellationTokenSource } from '../../src/utils/CancellationToken';
+
+describe('FakeDataSchemaService', () => {
+  describe('getEdgeTypes', () => {
+    it('returns non-empty array', async () => {
+      // Arrange
+      const service = new FakeDataSchemaService();
+
+      // Act
+      const types = await service.getEdgeTypes();
+
+      // Assert
+      assert.isAbove(types.length, 0);
+    });
+
+    it('caches data correctly', async () => {
+      // Arrange
+      const service = new FakeDataSchemaService();
+      const expected = await service.getEdgeTypes();
+
+      // Act
+      const types = await service.getEdgeTypes();
+
+      // Assert
+      assert.equal(types, expected);
+    });
+
+    it('returns entries with non empty name', async () => {
+      // Arrange
+      const service = new FakeDataSchemaService();
+
+      // Act
+      const types = await service.getEdgeTypes();
+
+      // Assert
+      for(const type of types) {
+        assert.equal(type.name, type.name.trim());
```

Comment:
put `assert.equal(type.name, type.name.trim());` into a different test, since one test should only test one thing.

Reply:
This runs in a loop, how do you want to put the assert into another tested? This only test a thing logically.

Reply:
yeah, if you want to test, whether the `type.name`'s have spaces at start or end put that into another test, otherwise just test `assert.isAbove(type.name.trim().length, 0)`.